instrumentation/spring/spring-webflux-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webflux/server/AdviceUtils.java

```diff
@@ -35,20 +28,10 @@ public static String spanNameForHandler(Object handler) {
     return className + ".handle";
   }
 
-  public static <T> Mono<T> setPublisherSpan(
-      Mono<T> mono, io.opentelemetry.context.Context context) {
-    return mono.<T>transform(finishSpanNextOrError(context));
-  }
-
-  /**
-   * Idea for this has been lifted from https://github.com/reactor/reactor-core/issues/947. Newer
-   * versions of reactor-core have easier way to access context but we want to support older
-   * versions.
-   */
-  public static <T> Function<? super Publisher<T>, ? extends Publisher<T>> finishSpanNextOrError(
-      io.opentelemetry.context.Context context) {
-    return Operators.lift(
-        (scannable, subscriber) -> new SpanFinishingSubscriber<>(subscriber, context));
+  public static <T> Mono<T> setPublisherSpan(Mono<T> mono, Context context) {
+    return mono.doOnError(t -> finishSpanIfPresent(context, t))
```

Comment:
Any thoughts on `doOnError` only setting the exception and `doOnTerminate` finishing the span?

Reply:
that doesn't really fit the Instrumenter model though, of calling end() and passing the exception?

Reply:
(Instrumenter coming in #3798)

Reply:
Ah that makes sense too